pkg/frontend/admin_ovsflows_get.go

```diff
@@ -0,0 +1,173 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func (f *frontend) getOVSFlows(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	// Main logic written in f._getOSVFlow() func
+	b, err := f._getOVSFlows(ctx, r, log)
+	if err != nil {
+		log.Error(err)
+	}
+	reply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getOVSFlows(ctx context.Context, r *http.Request, log *logrus.Entry) ([]byte, error) {
+	vars := mux.Vars(r)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	//Checking if the node mentioned in the query is correct or not.
+	nodeName := r.URL.Query().Get("node")
+	if nodeName == "" {
+		err := api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "Invalid Node Name")
+		return nil, err
+	}
+
+	// Getting OpenShiftCluster Document
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		log.Error(api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"]))
+		return nil, err
+	case err != nil:
+		return nil, err
+	}
+
+	// Getting rest config to create *kubernetes.Clientset
+	restConfig, err := restconfig.RestConfig(f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return nil, err
+	}
+	config, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Changed the code and added tests as well .. will be reflected in my new commit.

Reply:
@m1kola , actually .. Kubeactions does not have any method for extracting logs ... so the currently imported packages(client-go etc) they will still be needed to extract logs using the current setup.. and the same can be used to create the pod as well .. if we want to use kubeactions ... then a kubeinstance specially to create the pod would be need and currently .. it is being created without it only .. 
At first I thought of creating getLogs method for kubeaction only .. so that it can be re-used by others as well in future, but It looked like the operations are only CRUD operations and I thought kubeactions are meant to be like that .. 
What do you suggest here?

Reply:
@anshulvermapatel I'm not sure that I understnad your concerns. Let me clarify my comment just in case: I'm not expecting kubeactiosn to be used in the current state: it is probably not possible at the moment (e.g. no way to get logs, etc). But we can look into making modifications to enable what we need.

But before doing this, I think we need to take a step back and look into alternative options first. See my comment [here](https://github.com/Azure/ARO-RP/pull/1859#pullrequestreview-820288777).